### PR TITLE
Remove reliance on WebSocketChannel.new

### PR DESCRIPTION
**Pull request by brianquinlan**
- Thanks for your contribution! Please replace this text with a description of what this PR is changing or adding and why, list any relevant issues, and review the contribution guidelines below.

---

- [ ] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: build_daemon/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 4.0.2-wip
```
Comment (jakemac53) [inline build_daemon/CHANGELOG.md:1]: We could go ahead and publish right away, so I would drop `-wip` from the versions

Comment (brianquinlan) [reply build_daemon/CHANGELOG.md:1]: I was imaging adding this code to `package:web_socket_channel`, doing a release and then allowing `build_daemon` to support that version.

Is that too slow?

Comment (jakemac53) [reply build_daemon/CHANGELOG.md:1]: That should be fine, but we could just wait to land this at all then until that is ready?

Comment (brianquinlan) [reply build_daemon/CHANGELOG.md:1]: I don't have a strong opinion. So how about I make the changes that you suggested and land this PR. That will take some of the pressure off to land the change in `web_socket_channel` and publish it.

**Comment by rohanjariwala03** [comment]
@brianquinlan @jakemac53 Do you guys knows any estimate time for merging this PR and build newer version. As My build is failing due to difference between web_socket_channel version in my pubspec.yaml file and your's.

File: build_daemon/CHANGELOG.md (hunk 1 above)
Comment (jakemac53) [reply build_daemon/CHANGELOG.md]: SGTM

**Review by jakemac53 — APPROVED** [review]

**Comment by jakemac53** [comment]
> @brianquinlan @jakemac53 Do you guys knows any estimate time for merging this PR and build newer version? As My build is failing due to difference between web_socket_channel version in my pubspec.yaml file and your's.

I just released these so you should be good to go 🤞

**Pull request merged**